### ISSUE TITLE
Setting dotspacemacs-whitespace-cleanup to 'cleanup breaks magit

### ISSUE THREAD

**Issue by anlsh** [syl20bnr/spacemacs#13296]
#### Description :octocat:
Setting dotspacemacs-whitespace-cleanup to 'cleanup breaks magit

#### Reproduction guide :beetle:
- In the dotspacemacs/init function, change the value for dotspacemacs-whitespace-cleanup to from nil (default) to 'cleanup
- Start Emacs, open any magit status buffer in existing git repo: make some dummy changes to a file so that there's something to show in the status buffer

Note: You can also clone my .spacemacs from (https://github.com/anlsh/.spacemacs.d/tree/whitespace-breaks-magit)
in case there's something particular to my specific config causing the problem

#### *Observed behaviour:* :eyes: :broken_heart:
Faces are largely broken. Diffs appear fully expanded in white, without the usual green/red
faces for additions/deletions. Attempting to collapse the diffs via "TAB" gives some error
about not being able to collapse the root directory.

Otherwise the buffer seems to function (at least, I was able to stage and unstage changes).
It's possible that faces are also broken in other magit buffers like log, I didn't bother
investigating further.

Reverting dotspacemacs-whitespace-cleanup's value to nil fixes the buffer completely.

#### *Expected behaviour:* :heart: :smile:
Value of dotspacemacs-whitespace-cleanup shouldn't affect magit-status buffer.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.60
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. 26b8fe0c3)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion better-defaults command-log emacs-lisp erc git helm latex markdown org pdf-tools python scheme syntax-checking)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS PDUMPER LCMS2 GMP

**Comment by smile13241324** [comment]
You have a broken spacemacs config, see below excerpt from the current dotfile template, accessible via `SPC f e D`:
```elisp
   ;; Delete whitespace while saving buffer. Possible values are `all'
   ;; to aggressively delete empty line and long sequences of whitespace,
   ;; `trailing' to delete only the whitespace at end of lines, `changed' to
   ;; delete only whitespace for changed lines or `nil' to disable cleanup.
   ;; (default nil)
   dotspacemacs-whitespace-cleanup 'all
```
This sets the value to `all` setting it to `cleanup` is invalid and consequently your config will break.

Please retry with one of below settings and close this issue if it works.

**Comment by jr0cket** [comment]
@anlsh you issue may be similar to https://github.com/syl20bnr/spacemacs/issues/13216 in which case try deleting all the magit packages from the `.emacs.d/elpa` directory and restarting Emacs.

Magit upstream has gone through some refactoring and caused this kind of issue, although with new packages it is working correctly on Emacs 26.3

**Comment by anlsh** [comment]
Apologies, I somehow substituted `cleanup` everywhere I meant to write `changed`. I do believe that I must have tested with `changed` however, because I just checked and setting the (invalid) value of `cleanup` doesn't break magit.

Setting the value of `changed`, however, does produce output exactly like the screenshot in the issue @jr0cket linked (though oddly, not like I described above). Anyways I am on Emacs 27, so that would seem to be the issue
